default/AI/AIstate.py

```diff
@@ -120,8 +127,28 @@ def __setstate__(self, state_dict):
         for odict_attrib in ['colonisablePlanetIDs', 'colonisableOutpostIDs']:
             if dict_attrib not in state_dict:
                 self.__dict__[odict_attrib] = odict()
+        if 'uuid' not in state_dict:
+            self.uuid = uuid4()
+        if 'turn_hashes' not in state_dict:
+            self.turn_uuids = {}
         self.__dict__.setdefault('empire_standard_enemy_rating', 40)
 
+    def set_turn_uuid(self):
+        """
+        Set turn uuid. Should be called once per generateOrders
```

Comment:
Your only uses so far are trivial, but if you add nontrivial uses it seems an unnecessary complication/limitation that it could only be used in ways that are compatible with it being reset multiple times per turn, and that also seems somewhat inconsistent with the admonition that it be used once per GenerateOrders.  Why not do something like replacing the entire body with 
```
    return self.turn_uuids.setdefault(fo.currentTurn(), uuid4())
```